### PR TITLE
Documented design-time builds

### PR DESCRIPTION
**Pull request by davkean**
Wrote a document talking about design-time builds both for the existing and new project system.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/project-system

File: docs/design-time-builds.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+The design-time targets are typically are simple wrappers around their normal target equivalents, with customized behavior for design-time builds. 
+
+## Designing targets for use in design-time builds
+
+Targets that dynamically change references, source files or compilation options _must_ run during design-time builds to avoid unexpected behavior in Visual Studio. However, if a target does not contribute these items, then it should avoid running in these builds for performance reasons. Whether a target is run in design-time builds is controlled by a target's `BeforeTargets` and `AfterTargets` attributes and whether it specifies a direct or indirect dependency of the above targets. See [Diagnosing design-time builds](#Diagnosing-design-time-builds) to see logs that help you capture this.
+
+### Running in a design-time build
+
+If you've determined that your target needs to run in a design-time build, using the, using the above table, set `BeforeTargets` to the normal target equivalent of what you are contributing to the build. For example, if a target changes `<Reference>` items, then it should indicate that it runs _before_ `ResolveAssemblyReferences` target:
```
Comment (abpiskunov) [inline docs/design-time-builds.md]: extra 'using the'

**Comment by srivatsn** [comment]
👍

File: docs/design-time-builds.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+Design-Time Target                  | Normal Target                      | Description
+------------------------------------|------------------------------------|------------------
+ResolveAssemblyReferencesDesignTime | ResolveAssemblyReferences          | Resolves `<Reference>` items to their paths.
+ResolveProjectReferencesDesignTime  | ResolveProjectReferences           | Resolves `<ProjectReference>` items to their output paths.
+ResolveComReferencesDesignTime      | ResolveComReferences               | Resolves `<COMReference>` items to their primary interop assemblies (PIA) paths.
+Compile                             | Compile                            | Passes command-line arguments, include `<Compile>` and `<Analyzer>` items to the compiler[normal builds] or language service [design-time builds].
+
+The design-time targets are typically are simple wrappers around their normal target equivalents, with customized behavior for design-time builds.
```
Comment (basoundr) [inline docs/design-time-builds.md]: `are typically are` -> `typically are`

**Comment by basoundr** [comment]
:+1:

**Comment by jinujoseph** [comment]
:+1:

File: docs/design-time-builds.md (hunk 1 above)
Comment (davkean) [inline docs/design-time-builds.md]: Fixed

File: docs/design-time-builds.md (hunk 2 above)
Comment (davkean) [inline docs/design-time-builds.md]: Fixed.

**Pull request merged**